realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -818,11 +819,8 @@ private void checkIndex(int location) {
         }
     }
 
-    private void checkValidView() {
+    private void checkRealmValid() {
```

Comment:
Minor: `checkValidRealm` ?

Reply:
Also, why did we have the attached check later. Couldn't we still risk it happening inside `OsList` ?

Reply:
Object Store is checking valid of `LinkView` everywhere